From b0486ff2fd2912fa47b4c660829f3b891cfb3e99 Mon Sep 17 00:00:00 2001
From: Eberhard Beilharz <eb1@sil.org>
Date: Tue, 10 Apr 2012 11:51:30 +0200
Subject: [PATCH] Enable building with glib >= 2.32

The threads API changed in glib 2.32. Later versions of gtk-sharp have
adjusted to this, but 2.12.10 doesn't have these changes. This fix
checks the version of glib during configuration and includes the
deprecated version of the gthreads header if necessary.

glib2.32 is in Ubuntu Precise and Debian Unstable, so this change is
necessary for building gtk-sharp 2.12.10 on either of those systems.
---
 configure.in.in    |    2 ++
 glib/glue/list.c   |    5 +++++
 glib/glue/slist.c  |    5 +++++
 glib/glue/thread.c |   10 ++++++++++
 4 files changed, 22 insertions(+), 0 deletions(-)

diff --git a/configure.in.in b/configure.in.in
index a6dffd1..271fdfa 100644
--- a/configure.in.in
+++ b/configure.in.in
@@ -187,6 +187,8 @@ PKG_CHECK_MODULES(GLIB, gobject-2.0 >= $GTK_REQUIRED_VERSION)
 AC_SUBST(GLIB_CFLAGS)
 AC_SUBST(GLIB_LIBS)
 
+PKG_CHECK_EXISTS(gobject-2.0 >= 2.32, AC_DEFINE([GLIB_NEW_API], [1], [Glib has the new threads API]))
+
 PKG_CHECK_MODULES(PANGO, pango)
 AC_SUBST(PANGO_CFLAGS)
 AC_SUBST(PANGO_LIBS)
diff --git a/glib/glue/list.c b/glib/glue/list.c
index dd3f0de..aa8c912 100644
--- a/glib/glue/list.c
+++ b/glib/glue/list.c
@@ -19,8 +19,13 @@
  * Boston, MA 02111-1307, USA.
  */
 
+#include "config.h"
 
+#ifdef GLIB_NEW_API
+#include <glib.h>
+#else
 #include <glib/glist.h>
+#endif
 
 /* Forward declarations */
 gpointer gtksharp_list_get_data (GList *l);
diff --git a/glib/glue/slist.c b/glib/glue/slist.c
index 80f5f17..3cf81e4 100644
--- a/glib/glue/slist.c
+++ b/glib/glue/slist.c
@@ -19,8 +19,13 @@
  * Boston, MA 02111-1307, USA.
  */
 
+#include "config.h"
 
+#ifdef GLIB_NEW_API
+#include <glib.h>
+#else
 #include <glib/gslist.h>
+#endif
 
 /* Forward declarations */
 gpointer gtksharp_slist_get_data (GSList *l);
diff --git a/glib/glue/thread.c b/glib/glue/thread.c
index 4fcf8c6..4416d12 100644
--- a/glib/glue/thread.c
+++ b/glib/glue/thread.c
@@ -19,14 +19,24 @@
  * Boston, MA 02111-1307, USA.
  */
 
+#include "config.h"
 
+#ifdef GLIB_NEW_API
+#include <glib.h>
+#else
 #include <glib/gthread.h>
+#endif
 
 gboolean glibsharp_g_thread_supported (void);
 
 gboolean
 glibsharp_g_thread_supported ()
 {
+#ifdef GLIB_NEW_API
+	// this mimicks what Ubuntu does in the C# glue code
+	return TRUE;
+#else
 	return g_thread_supported ();
+#endif
 }
 
-- 
1.7.5.4

